Python/pywarpx/_libwarpx.py

```diff
@@ -414,15 +424,25 @@ def add_particles(species_name,
         uy = np.array(uy)*np.ones(maxlen)
     if lenuz == 1:
         uz = np.array(uz)*np.ones(maxlen,'d')
-    if lenattr == 1:
-        nattr = get_nattr()
-        attr = np.array(attr)*np.ones([maxlen,nattr])
+    for key, val in kwargs.items():
+        if np.size(val) == 1:
+            kwargs[key] = np.array(val)*np.ones(maxlen)
+
+    nattr = get_nattr_species(species_name) - 3
```

Comment:
Documenting magic constants now here:
```suggestion
    # --- The -3 is because the comps include the velocites
    nattr = get_nattr_species(species_name) - 3
```